### ISSUE TITLE
Static library support

### ISSUE THREAD

**Issue by mingw-io** [open-quantum-safe/oqs-provider#82]
Hi.

Many people build OpenSSL as a static library.
It appears that OQS provider does NOT work properly with a static OpenSSL as some failing tests demonstrate (e.g. oqs_endecode).
We have been experimenting/trying to build OpenSSL statically with OQS provider 'embedded' the same way as the legacy provider is built.
This is our first attempt and it seems to be working.
Obviously there are advantages and disadvantages here.
The main advantage is that we end up with only one file.

We have run all OQS tests (oqs_signatures, oqs_kems, oqs_groups & oqs_endecode) and they have all passed.

Do we need to run/do more testing here? We can make the binary available (if that helps!).

Cheers and thanks for this project.

![image](https://user-images.githubusercontent.com/70483213/198078367-ed64cc5a-80f5-42a5-8b50-12d0b0795c3e.png)

**Comment by baentsch** [comment]
> We can make the binary available (if that helps!).

The approach to "build static" sounds good -- what would help me is a (windows?) script building everything as you recommend, triggering testing (both good and failing), ideally using gnu-type tooling. A pure binary is something I wouldn't know how to debug into (which tools to use as I assume they depend on your build environment, right?).

**Comment by mingw-io** [comment]
Some (extra) testing: client/server

![image](https://user-images.githubusercontent.com/70483213/198333158-8c4f11a6-5b52-4e7e-b7d6-981b44e5b7d6.png)


![image](https://user-images.githubusercontent.com/70483213/198333204-0c60fa6b-f0ec-4eb1-9018-190739742354.png)

**Comment by baentsch** [comment]
> On Windows, we can use the same toolchain we use on Mac & Linux (a.k.a. GNU GCC & cmake).

Just to make it clear for a simple mind like me: I need to download and install all software from https://sourceforge.net/projects/mingw/, then run `cmake -DCMAKE_INSTALL_PREFIX=<some folder here> .. && cmake --build`? No need to add anything along the lines of 
```
set(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
set(BUILD_SHARED_LIBS OFF)
set(CMAKE_EXE_LINKER_FLAGS "-static")
```
?

**Comment by mingw-io** [comment]
Hi.
Find attached a OneClick installer for testing/verification purposes. We hope you find it useful.

![image](https://user-images.githubusercontent.com/70483213/198562922-22cc8388-b597-4318-b4fc-fdf78df7839d.png)




[oqs-provider-0.4.0-x64.zip](https://github.com/open-quantum-safe/oqs-provider/files/9887363/oqs-provider-0.4.0-x64.zip)

**Comment by baentsch** [comment]
Thanks very much for this installer. I am now confused by the introductory statement, though: 

> as some failing tests demonstrate (e.g. oqs_endecode

When trying out the installer and the test programs contained, no test fails (?). So did you manage to resolve all problems? If so, would you consider contributing your work e.g., to resolve this issue, #80 and #47 ?

**Comment by baentsch** [comment]
@mingw-io ping/Did you see the questions above?

**Comment by mingw-io** [comment]
Ciao there.
It is up to this project maintainer(s) to make up their minds about how to proceed/move forward!

Regarding issue #80 my tests have demonstrated the oqs-provider does NOT support OpenSSL statically linked.
This issue (#82) is related to #80. 
All issues (#47, #80 & this one) are resolved on my end.

Now we need to ask ourselves how to proceed here.
I have noticed that some/many (end) users have difficulties in either building OQS from source and/or setting it up and/or using it.

I could contribute some binaries/installer with no dependencies at all (e.g. Visual Studio, etc).
The installer provided above contains three oqs official tests. When you run them, all should pass. If not, please let me know.

**Comment by baentsch** [comment]
> I could contribute some binaries/installer with no dependencies at all (e.g. Visual Studio, etc).

PR would be welcome.

> The installer provided above contains three oqs official tests. When you run them, all should pass. If not, please let me know.

I do not have a Windows installation to test. We'd expect any contribution to contain CI features (for the specific platform, i.e., Windows in this case) such as to not depend on manual testing.

**Comment by baentsch** [comment]
#201 seems to resolve this issue.

**Comment by baentsch** [comment]
Resolved by #201
